mne/simulation/raw.py

```diff
@@ -0,0 +1,511 @@
+# Authors: Mark Wronkiewicz <wronk@uw.edu>
+#          Yousra Bekhti <yousra.bekhti@gmail.com>
+#          Eric Larson <larson.eric.d@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import warnings
+from copy import deepcopy
+
+from .evoked import _generate_noise
+from ..event import _get_stim_channel
+from ..io.pick import pick_types, pick_info, pick_channels
+from ..source_estimate import VolSourceEstimate
+from ..cov import make_ad_hoc_cov, read_cov
+from ..bem import fit_sphere_to_headshape, make_sphere_model, read_bem_solution
+from ..io import RawArray, _BaseRaw, get_chpi_positions, FIFF
+from ..forward import (_magnetic_dipole_field_vec, _merge_meg_eeg_fwds,
+                       _stc_src_sel, convert_forward_solution,
+                       _prepare_for_forward, _prep_meg_channels,
+                       _compute_forwards, _to_forward_dict)
+from ..transforms import _get_mri_head_t, transform_surface_to
+from ..source_space import read_source_spaces, _points_outside_surface
+from ..source_estimate import _BaseSourceEstimate
+from ..utils import logger, verbose, check_random_state
+from ..externals.six import string_types
+
+
+@verbose
+def simulate_raw(raw, stc, trans, src, bem, cov='simple',
+                 blink=False, ecg=False, chpi=False, head_pos=None,
+                 mindist=1.0, interp='cos2', n_jobs=1, random_state=None,
+                 verbose=None):
+    """Simulate raw data with head movements
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw template to use for simulation. The ``info``, ``times``,
+        and potentially ``first_samp`` properties will be used.
```

Comment:
You'd need to pass 1) `info`, 2) `times`, 3) `first_samp`, and 4) replicate `raw.time_as_index` to get the same functionality as now, so I'm not sure it's worth it. As it is now, we can eventually also make it so `head_pos='estimated'` where it estimates the head positions from raw using cHPI information, whereas by not passing `raw` we'd have to do it in separate steps.